pkg/controller/master/virtualmachine/vmi_controller.go

```diff
@@ -109,3 +111,48 @@ func (h *VMIController) removeDeprecatedFinalizer(_ string, vmi *kubevirtv1.Virt
 	}
 	return vmi, nil
 }
+
+// PatchMacAddresses patch mac addresses to virtual machine when virtual machine instance is deleted
```

Comment:
This function targets to restore the saved MAC on VM's annotation to VMI's interface, then it should be put in VMI's mutator, where when VMI is created then patch them;  

when controller does it, it will race with kubevirt